File: invesdwin-context-parent/invesdwin-context-clojure/src/test/java/de/invesdwin/context/clojure/tests/InputsAndResultsTestDouble.clj
(defmacro defined [& var]
    `(try 
        (or 
            (boolean (resolve ~@var)) 
            (not (nil? (eval ~@var)))
        ) (catch Exception e# false)
    )
)

(println "getDouble")
(if (defined 'getDouble)
	(throw (Exception. "getDouble already defined!"))
)
(def getDouble putDouble)
(def getDoubleType (class getDouble))
(println getDoubleType)
(println getDouble)
(if-not (= getDoubleType Double)
	(throw (Exception. "getDouble not Double!"))
)

(println "getDoubleVector")
(if (defined 'getDoubleVector)
	(throw (Exception. "getDoubleVector already defined!"))
)
(def getDoubleVector putDoubleVector)
(def getDoubleVectorType (class (get getDoubleVector 0)))
(println getDoubleVectorType)
(println getDoubleVector)
(if-not (= getDoubleVectorType Double)
	(throw (Exception. "getDoubleVector not Double!"))
)

(println "getDoubleVectorAsList")
(if (defined 'getDoubleVectorAsList)
	(throw (Exception. "getDoubleVectorAsList already defined!"))
)
(def getDoubleVectorAsList putDoubleVectorAsList)
(def getDoubleVectorAsListType (class (get getDoubleVectorAsList 0)))
(println getDoubleVectorAsListType)
(println getDoubleVectorAsList)
(if-not (= getDoubleVectorAsListType Double)
	(throw (Exception. "getDoubleVectorAsList not Double!"))
)

(println "getDoubleMatrix")
(if (defined 'getDoubleMatrix)
	(throw (Exception. "getDoubleMatrix already defined!"))
)
(def getDoubleMatrix putDoubleMatrix)
(def getDoubleMatrixType (class (get (get getDoubleMatrix 0) 0)))
(println getDoubleMatrixType)
(println getDoubleMatrix)
(if-not (= getDoubleMatrixType Double)
	(throw (Exception. "getDoubleMatrix not Double!"))
)

(println "getDoubleMatrixAsList")
(if (defined 'getDoubleMatrixAsList)
	(throw (Exception. "getDoubleMatrixAsList already defined!"))
)
(def getDoubleMatrixAsList putDoubleMatrixAsList)
(def getDoubleMatrixAsListType (class (get (get getDoubleMatrixAsList 0) 0)))
(println getDoubleMatrixAsListType)
(println getDoubleMatrixAsList)
(if-not (= getDoubleMatrixAsListType Double)
	(throw (Exception. "getDoubleMatrixAsList not Double!"))
)

File: invesdwin-context-parent/invesdwin-context-scala/src/test/java/de/invesdwin/context/scala/tests/InputsAndResultsTestDouble.scala
println("getDouble")
if(getDouble != null)
	throw new Exception("getDouble already defined!")
val getDouble = putDouble
val getDoubleType = getDouble.getClass()
println(getDoubleType)
println(getDouble)
if(getDoubleType != classOf[java.lang.Double])
	throw new Exception("getDouble not Double!")

println("getDoubleVector")
if(getDoubleVector != null)
	throw new Exception("getDoubleVector already defined!")
val getDoubleVector = putDoubleVector.asInstanceOf[Array[Double]]
val getDoubleVectorType = getDoubleVector(0).getClass()
println(getDoubleVectorType)
println(getDoubleVector)
if(getDoubleVectorType != classOf[Double])
	throw new Exception("getDoubleVector not Double!")

println("getDoubleVectorAsList")
if(getDoubleVectorAsList != null)
	throw new Exception("getDoubleVectorAsList already defined!")
val getDoubleVectorAsList = putDoubleVectorAsList.asInstanceOf[Array[Double]]
val getDoubleVectorAsListType = getDoubleVectorAsList(0).getClass()
println(getDoubleVectorAsListType)
println(getDoubleVectorAsList)
if(getDoubleVectorAsListType != classOf[Double])
	throw new Exception("getDoubleVectorAsList not Double!")

println("getDoubleMatrix")
if(getDoubleMatrix != null)
	throw new Exception("getDoubleMatrix already defined!")
val getDoubleMatrix = putDoubleMatrix.asInstanceOf[Array[Array[Double]]]
val getDoubleMatrixType = getDoubleMatrix(0)(0).getClass()
println(getDoubleMatrixType)
println(getDoubleMatrix)
if(getDoubleMatrixType != classOf[Double])
	throw new Exception("getDoubleMatrix not Double!")

println("getDoubleMatrixAsList")
if(getDoubleMatrixAsList != null)
	throw new Exception("getDoubleMatrixAsList already defined!")
val getDoubleMatrixAsList = putDoubleMatrixAsList.asInstanceOf[Array[Array[Double]]]
val getDoubleMatrixAsListType = getDoubleMatrixAsList(0)(0).getClass()
println(getDoubleMatrixAsListType)
println(getDoubleMatrixAsList)
if(getDoubleMatrixAsListType != classOf[Double])
	throw new Exception("getDoubleMatrixAsList not Double!")
